src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java

```diff
@@ -295,6 +312,49 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
         }
     }
 
+    /**
+     * Get content from request body and add the attachment
+     *
+     * @param multipart Multi part body the messages should be added to
+     * @param attachment  the attachment
+     * @throws MessagingException if failed to set attachments
+     * @throws IOException        if an error occurred while reading attachment content
+     */
+    private void addAttachment(MimeMultipart multipart, JSONObject attachment) throws MessagingException, IOException {
+
+        String attachmentName;
+        try {
+            attachmentName = (String) attachment.get(EmailConstants.ATTACHMENT_FILE_NAME);
+            MimeBodyPart part = new MimeBodyPart();
+            String fileContent = (String) attachment.get(EmailConstants.ATTACHMENT_FILE_CONTENT);
+            try (InputStream bain = new ByteArrayInputStream(isBase64(fileContent) ?
+                    Base64.decodeBase64(fileContent.getBytes()) : fileContent.getBytes())) {
+                part.setDisposition(ATTACHMENT);
+                part.setFileName(attachmentName);
+                Tika tika = new Tika();
+                String fileContentType = tika.detect(bain);
+                DataHandler dataHandler = new DataHandler(new EmailAttachmentDataSource(attachmentName, bain,
+                        fileContentType));
+                part.setDataHandler(dataHandler);
+                part.setHeader(CONTENT_TYPE_HEADER, dataHandler.getContentType());
+                part.setHeader(CONTENT_TRANSFER_ENCODING_HEADER, this.contentTransferEncoding);
+                multipart.addBodyPart(part);
+            }
+        } catch (JSONException e) {
+            throw new IOException("Invalid JSON data format. " + e.getMessage());
+        }
+    }
+
+    private boolean isBase64(String stringBase64){
+        String regex = "([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)";
+        Pattern pattern = Pattern.compile(regex);
+        if (!pattern.matcher(stringBase64).matches()) {
+            return false;
+        } else {
+            return true;
+        }
```

Comment:
shall we simplify this to return pattern.matcher(stringBase64).matches();

Reply:
Done